### PR TITLE
Lint using putout part 3

### PR DESCRIPTION
**Pull request by coderaiser**
After successfully merged linting sessions [part1](https://github.com/xtermjs/xterm.js/pull/2953) and [part2](https://github.com/xtermjs/xterm.js/pull/3133), the time is come for part 3 :).

Any rule can be disabled.
Command used:

```sh
putout {src,addons}/**/*.ts --fix
```

Applied rules:
- [remove-useless-conditions](https://github.com/coderaiser/putout/tree/v16.0.0/packages/plugin-remove-useless-conditions)
- [apply-optional-chaining](https://github.com/coderaiser/putout/tree/v16.0.0/packages/plugin-apply-optional-chaining)
- [remove-useless-types-from-constants](https://github.com/coderaiser/putout/tree/v16.0.0/packages/plugin-remove-useless-types-from-constants)
- [convert-for-each-to-for-of](https://github.com/coderaiser/putout/tree/v16.0.0/packages/plugin-convert-for-each-to-for-of)
- [extract-sequence-expressions](https://github.com/coderaiser/putout/tree/v16.0.0/packages/plugin-extract-sequence-expressions)
- [regexp/optimize](https://github.com/coderaiser/putout/tree/v16.0.0/packages/plugin-regexp#regexpoptimize)
- [convert-index-of-to-includes](https://github.com/coderaiser/putout/tree/v16.0.0/packages/plugin-promises#addmissingawait)
- [promises/add-return-await](https://github.com/coderaiser/putout/tree/v16.0.0/packages/plugin-promises#addreturnawait)
- [apply-shorthand-properties](https://github.com/coderaiser/putout/tree/v16.0.0/packages/plugin-apply-shorthand-properties)

Current config for [putout v16](https://github.com/coderaiser/putout):

```json
{
    "rules": {
        "strict-mode": "off",
        "remove-useless-spread/object": "off",
        "remove-iife": "off",
        "convert-generic-to-shorthand": "off",
        "convert-assignment-to-comparison": "off",
        "remove-console": "off",
        "remove-unused-variables": "off",
        "remove-useless-variables": "off",
        "apply-destructuring": "off",
        "merge-if-statements": "off",
        "convert-apply-to-spread": "off",
        "convert-math-pow": "off",
        "convert-for-to-for-of": "off",
        "convert-template-to-string": "off",
        "remove-empty": "on",
        "convert-index-of-to-includes": "on",
        "convert-for-each-to-for-of": "on",
        "apply-shorthand-properties": "on",
        "apply-numeric-separators": "off",
        "promises/add-missing-await": "off"
    },
    "match": {
        "SerializeAddon.benchmark.ts": {
            "remove-unused-expressions": "off"
        }
    },
    "plugins": [
        "apply-shorthand-properties"
    ]
}
```

### CODE REVIEW

**Review by jerch — APPROVED** [review]
Great work, your tools seems quite capable :+1: . Nothing to mock about from my side.

**Review by Tyriar — APPROVED** [review]
Looks good, love the `for ... of` changes as I've come to dislike `forEach` greatly 🙂

**Comment by coderaiser** [comment]
> Great work, your tools seems to be quite capable 👍 . Nothing to mock about from my side.

Thank you :), I'm using it a lot for all my projects, and want it to be useful for community as much as possible :).

> Edit:
Ok lemme re-phrase that - your tool seems to make a really great job in transpiling and finding code smells/errors. Maybe you should think about promoting it to other tools like linters/babel, so they can directly benefit from its capabilities?

`putout` uses `babel` from the inside, and it benefit a lot from `babel` and [@babel/template](https://babeljs.io/docs/en/babel-template) and move code transforming to the next level, using custom language fully compatible with `JavaScript` with help of [@putout/compare](https://github.com/coderaiser/putout/tree/v16.1.0/packages/compare), like in `remove-useless-conditions`:

```js
module.exports.replace = () => ({
    'if (__a?.__b) {__a.__b(__args)}': '__a?.__b(__args)',
});
```

Also it calls `eslint` after `transforming` is done.

So, because `eslint` uses [estree AST](https://github.com/estree/estree), they unlikely will use `babel tooling` which uses [babel AST](https://github.com/babel/babel/blob/main/packages/babel-parser/ast/spec.md). 
Guys from [rome](https://rome.tools/) writes everything from scratch, and using same `visitor traversing` which is good for speed, good for speed but challenging for support.

Also `putout` has a `progress bar`, and support for `md`, `yaml`, `html`, `css`, `json` files from the box,  because integrations with other tooling is top priority.

Anyways for `babel` it can be useful, because of simplified transformations of the code, so this is a good idea and I will be glad for any help :). From the first day `putout` written as `Open Source` solution divided on independent packages to make code transforming simple, popular and available to everyone.

> Looks good, love the for ... of changes as I've come to dislike forEach greatly 

Nice :), same with me, `for-of` looks much cleaner

File: addons/xterm-addon-fit/test/FitAddon.api.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ describe('FitAddon', () => {
 
   describe('proposeDimensions', () => {
     afterEach(async () => {
-      return unloadFit();
+      return await unloadFit();
```
[the remark this answers is not in the document]
Comment (coderaiser) [reply addons/xterm-addon-fit/test/FitAddon.api.ts:48]: Yes :), it's located in [@putout/promises](https://github.com/coderaiser/putout/tree/master/packages/plugin-promises#add-return-await). What it does is looking for declarations of `functions` and if it is `async`, it adds `await`.

**Comment by Tyriar** [comment]
> Nice :), same with me, for-of looks much cleaner

The main reason for me is debugging `forEach` is a pain since it uses callback

**Pull request merged**